lib/analyser.py

```diff
@@ -3,3 +3,134 @@
+        7.2. growth capex = (New fixed asset - fixed asset)
+        7.3. Add maintainence capex of 4% of fixed assets             
+        7.4. capex = (New fixed asset - fixed asset) + 0.04*fixed asset
+        7.5. update the fixed asset with new capex
+    8. market cap
+    9. no of years = real holding time of the investment
+    """
+    # Initializations
+    net_present_value   = 0
+    profit              = init_profit       #base profit
+    fixed_asset         = init_fixed_asset  #base fixed asset
+    capex               = 0
+
+    # Linear decay factor 
+    decay_factor = (init_profit_growth_rate - perpetual_growth_rate)/years_until_decay
+
+    for year in range(0, no_of_years):
+        #Calculate the profit growth & capex for each year
+        if(year<years_until_decay):
+            pgr = init_profit_growth_rate - (decay_factor*year)
+            capex = (fixed_asset*(pgr - perpetual_growth_rate + 0.04))
+        else:
+            pgr = perpetual_growth_rate
+            capex = fixed_asset*0.04
+        
+        #calculate the profit and the fixed asset value
+        profit = profit*(1+pgr)
+        fixed_asset = fixed_asset + capex
+        #Calculate the discounted value of cash after capex spending
+        discounted_profit = (profit-capex)/(pow((1+discount_rate),(year+1)))
+        net_present_value += discounted_profit
+
+    terminal_value = profit*(1+perpetual_growth_rate)/(discount_rate-perpetual_growth_rate)
+    discounted_terminal_value = terminal_value/(pow((1+discount_rate),(year+1)))
+    net_present_value += discounted_terminal_value
+    return (market_cap/net_present_value)
+
+def DCF_MC_analysis(self, init_profit, init_fixed_asset, market_cap):
+    """
+    Usage:
+    Margin of safety parameter
+    Provides the (mean+3*sigma) limit of the market_cap/net_present_value estimates.
+    This limit < 0.9 for enough margin of safety.
+
+    Params:
+    1. init_profit = {CFO, (OP - Tax paid) (this is better)}
+    """
+    # TODO : run tests on iterations to see what value is good enough
+    iterations              = 100000
+    init_profit_growth_rate = np.random.normal(loc=0.15, scale=0.01, size=iterations)
+    perpetual_growth_rate   = [0.05, 0.06]
+    years_until_decay       = [5, 6, 7, 8]
+    discount_rate           = np.random.normal(loc=0.12, scale=0.01, size=iterations)
+    no_of_years             = 10
+    npv_distribution        = []
+    # TODO : Try to add a dataframe based operation here to check if it is faster
+    for yud in years_until_decay:
+        for pgr in perpetual_growth_rate:
+            for j in range(iterations):
+                npv_distribution.append(self.calc_NPV(init_profit, init_profit_growth_rate[j], \
+                pgr, yud, discount_rate[j], init_fixed_asset, \
+                market_cap, no_of_years)) 
+    return np.mean(npv_distribution)+3*np.std(npv_distribution)
+
+def Calc_SSGR_revenue(revenue, NPM, DPR, fixed_assets_start, depreciation):
```

Comment:
```suggestion
def calc_ssgr_revenue(revenue, npm, dpr, fixed_assets_start, depreciation):
```

Reply:
Followed the standard convention and implemented the changes